docs/polkadot-kusama-bridge-overview.md

```diff
@@ -0,0 +1,132 @@
+  will pay the full cost for submitting all other headers;
+
+- we do not reward relayers for submitting parachain finality transactions. The relayer will pay the full cost for
+  submitting parachain finality transactions;
+
+- we compensate the cost of message delivery transactions that have actually delivered the messages. So if your
+  transaction has claimed to deliver messages `[42, 43, 44]`, but, because of some reasons, has actually delivered
+  messages `[42, 43]`, the transaction will be free for relayer. If it has not delivered any messages, then
+  the relayer pays the full cost of the transaction;
+
+- we compensate the cost of message delivery and all required finality calls, if they are part of the same
+  [`frame_utility::batch_all`](https://github.com/paritytech/substrate/blob/891d6a5c870ab88521183facafc811a203bb6541/frame/utility/src/lib.rs#L326)
+  transaction. Of course, the calls inside the batch must be linked - e.g. the submitted parachain head must be used
+  to prove messages. Relay header must be used to prove parachain head finality. If one of calls fails, or if they
+  are not linked together, the relayer pays the full transaction cost.
+
+Please keep in mind that the fee of "zero-cost" transactions is still withdrawn from the relayer account. But the
+compensation is registered in the `pallet_bridge_relayers::RelayerRewards` map at the target bridge hub. The relayer
+may later claim all its rewards later, using the `pallet_bridge_relayers::claim_rewards` call.
+
+*A side note*: why we don't simply set the cost of useful transactions to zero? That's because the bridge has its cost.
+If we won't take any fees, it would mean that the sender is not obliged to pay for its messages. And Bridge Hub
+collators (and, maybe, "treasury") are not receiving any payment for including transactions. More about this later,
+in the [Who is Rewarding Relayers](#who-is-rewarding-relayers) section.
+
+### Message Delivery Confirmation Rewards
+
+In addition to the "zero-cost" message delivery transactions, the relayer is also rewarded for:
+
+- delivering every message. The reward is registered during delivery confirmation transaction at the Source Bridge
+  Hub.;
+
+- submitting delivery confirmation transaction. The relayer may submit delivery confirmation that e.g. confirms
+  delivery of four messages, of which the only one (or zero) messages is actually delivered by this relayer. It
+  receives some fee for confirming messages, delivered by other relayers.
+
+Both rewards may be claimed using the `pallet_bridge_relayers::claim_rewards` call at the Source Bridge Hub.
+
+### Who is Rewarding Relayers
+
+Obviously, there should be someone who is paying relayer rewards. We want bridge transactions to have a cost, so we
+can't use fees for rewards. Instead, the parachains, that are using the bridge, have a mutual agreement, that they
+are paying each other costs.
```

Comment:
With the sovereign accounts model, I don't think there is the mutual agreement anymore. Like you say below, Statemine will pay _all_ relayer rewards for messages coming from Statemine (it pays on each side from its sov account on that respective BH). Statemint doesn't have to pay anything to receive messages from Statemine, right?

At most, Statemint might offer free XCM execution for messages with Computed Origin == Statemine, but in terms of relayer rewards it is doing nothing, right?

How about:
```suggestion
can't use fees for rewards. Instead, the parachains using the bridge, use sovereign accounts on both sides
of the bridge to cover relayer rewards.
```
?